client/release/scripts/prep-release.sh

```diff
@@ -0,0 +1,141 @@
+#!/bin/bash
+
+# Helper script to prepare a release for the JS SDK.
+
+# Read the build number from version-build to compare with new build number and ensure update has been made.
+output_version_file="client/release/version.js"
+version_template_file="client/release/templates/version.tpl.js"
+
+# Extracts the current build number for comparison 
+current_build=$(awk -F "['\"]" '/SDK_BUILD_NUMBER =/{print $2}' "${output_version_file}" | tr -d '\n')
+current_version=$(awk -F "['\"]" '/SDK_VERSION =/{print $2}' "${output_version_file}" | tr -d '\n')
+current_core_version=$(< client/release/version-sdk-core)
+
+version_sdk_core_file="client/release/version-sdk-core"
+
+core_modified="${1}"
+
+# Function to execute upon exit
+cleanup() {
+    echo "Performing cleanup tasks..."
+    # Revert changes to file if any
+    sed -e "s/{{ build }}/$current_build/" -e "s/{{ version }}/$current_version/" "$version_template_file" > "$output_version_file"
+    if [ "${core_modified}" = "true" ]; then
+        echo "${current_core_version}" > "${version_sdk_core_file}"
+    fi
+    exit 1   
+}
+
+# Set the trap to call the cleanup function on exit
+trap cleanup SIGINT
+
+enforce_latest_code() {
+    if [[ -n "$(git status --porcelain=v1)" ]]; then
+        echo "ERROR: working directory is not clean."
+        echo "Please stash your changes and try again."
+        exit 1
+    fi
+}
+
+# Function to validate the version number format x.y.z(-beta.w)
+update_and_validate_version() {
+    if [ "${core_modified}" = "true" ]; then
+        while true; do
+            # Prompt the user to input the version number
+            read -p "Enter the core version number (format: x.y.z(-beta.w)): " version
+            # Validate the version number format
+            if [[ "${version}" =~ ^[0-9]+\.[0-9]+\.[0-9]+(-beta\.[0-9]+)?$ ]]; then        
+                if [[ "${current_core_version}" != "${version}" ]]; then
+                    # TODO: Check the less than case as well.
+                    echo "${version}" > "${version_sdk_core_file}"
+                    echo "New version number is: ${version}"
+                    break
+                else
+                    echo "Core version hasn't changed."
+                fi        
+            else
+                echo "Invalid version number format: ${version}"
+                echo "Please enter a version number in the 'x.y.z(-beta.w)' format."
+            fi
+        done
+    fi
+    while true; do
+        # Prompt the user to input the version number
+        read -p "Enter the version number (format: x.y.z(-beta.w)): " version_publish
```

Comment:
```suggestion
        read -p "Enter the JS SDK client's version number (format: x.y.z(-beta.w)): " version_publish
```
To better distinguish from the other version number that's being asked

Reply:
Done